Common/XGBoost/src/main/java/org/tribuo/common/xgboost/XGBoostFeatureImportance.java

```diff
@@ -0,0 +1,253 @@
+package org.tribuo.common.xgboost;
+
+import ml.dmlc.xgboost4j.java.Booster;
+import ml.dmlc.xgboost4j.java.XGBoostError;
+import org.tribuo.ImmutableFeatureMap;
+
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static ml.dmlc.xgboost4j.java.Booster.FeatureImportanceType.*;
+
+/**
+ * Generate and collate feature importance information from the XGBoost model. This wraps the underlying functionality
+ * of the XGBoost model, and should provide feature importance metrics compatible with those provided by XGBoost's R
+ * and Python APIs. For a more treatment of what the different importance metrics mean and how to interpret them, see
+ * <a href="https://xgboost.readthedocs.io/en/latest/R-package/discoverYourData.html">here</a>. In brief
+ *
+ * <ul>
+ *     <li><b>Gain</b> measures the improvement in accuracy that a feature brings to the branches on which it appears.
+ *     This represents the sum of situated marginal contributions that a given feature makes to the each branching
+ *     chain in which it appears.</li>
+ *     <li><b>Cover</b> measures the number of examples a given feature discriminates across, relative to the total
+ *     number of examples all features discriminate across.</li>
+ *     <li><b>Weight</b> measures the number a times a feature occurs in the model. Due to the way the model builds trees,
+ *     this value is skewed in favor of continuous features.</li>
+ *     <li><b>Total Gain</b> is similar to gain, but not locally averaged by weight, and thus not skewed in the way that
+ *     weight can be skewed.</li>
+ *     <li><b>Total Cover</b> is similar to cover, but not locally averaged by weight, and thus not skewed in the way that
+ *     weight can be skewed.</li>
+ * </ul>
+ */
+public class XGBoostFeatureImportance {
+
+    /**
+     * An instance of feature importance values for a single feature. See {@link XGBoostFeatureImportance} for details
+     * on interpreting the metrics.
+     */
+    public static class XGBoostFeatureImportanceRecord {
```

Comment:
I'm dubious about giving this a name including "Record" given it's not a record yet, and calling records "somethingRecord" sounds a bit odd.

Reply:
How about `XGBoostFeatureImportanceInstance`?

Reply:
Sounds good.